libcni/api.go

```diff
@@ -233,6 +242,99 @@ func (c *CNIConfig) GetNetworkCachedResult(net *NetworkConfig, rt *RuntimeConf)
 	return getCachedResult(net.Network.Name, net.Network.CNIVersion, rt)
 }
 
+func stringifyArgs(args [][2]string) string {
+	entries := make([]string, len(args))
+	for i, kv := range args {
+		entries[i] = strings.Join(kv[:], "=")
+	}
+	return strings.Join(entries, ";")
+}
+
+func parseArgs(args string) ([][2]string, error) {
+	var result [][2]string
+
+	pairs := strings.Split(args, ";")
+	for _, pair := range pairs {
+		kv := strings.Split(pair, "=")
+		if len(kv) != 2 || kv[0] == "" || kv[1] == "" {
+			return nil, fmt.Errorf("invalid CNI_ARGS pair %q", pair)
+		}
+
+		result = append(result, [2]string{kv[0], kv[1]})
+	}
+
+	return result, nil
+}
+
+type cachedConfig struct {
```

Comment:
For future compatability, this should be something of the form

```json
{
  "kind": "CniCacheV1",
  "value": {"cniVersion": "..."  }
}
```

Reply:
@squeed Good point.